src/main/java/org/zalando/nakadi/service/WebResult.java

```diff
@@ -0,0 +1,21 @@
+package org.zalando.nakadi.service;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.zalando.problem.spring.web.advice.Responses;
+
+import java.util.function.Supplier;
+
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.ResponseEntity.status;
+
+public class WebResult {
+
+    public static ResponseEntity<?> wrap(final Supplier<Result> supplier, final NativeWebRequest request) {
```

Comment:
I don't really like these Result and WebResult.
In more complex cases (when we not just return 200) it will not work and we will anyway have to write code in controller.
And I don't really understand the desire to have 1-liners in controller methods.